osutil/disks/disks_linux.go

```diff
@@ -632,6 +649,54 @@ func (d *disk) populatePartitions() error {
 			}
 			part.KernelDeviceNode = devname
 
+			// we should always have the partition type
+			partType, ok := udevProps["ID_PART_ENTRY_TYPE"]
+			if !ok {
+				return fmt.Errorf("cannot get udev properties for device %s (a partition of %s), missing required udev property \"ID_PART_ENTRY_TYPE\"", partDev, d.Dev())
+			}
+
+			// on dos disks, the type is formatted like "0xc", when we prefer to
+			// always use "0C" so fix the formatting
+			if d.schema == "dos" {
+				partType = strings.TrimPrefix(partType, "0x")
+				v, err := strconv.ParseUint(partType, 16, 8)
+				if err != nil {
+					return fmt.Errorf("cannot get udev properties for device %s (a partition of %s), cannot convert MBR partition type %q: %v", partDev, d.Dev(), partType, err)
+				}
+				partType = fmt.Sprintf("%02X", v)
+			} else {
+				// on GPT disks, just capitalize the partition type since it's a
+				// UUID
+				partType = strings.ToUpper(partType)
+			}
+
+			part.PartitionType = partType
+
+			// the partition may not have a filesystem, in which case this might
+			// be the empty string
+			part.FilesystemType = udevProps["ID_FS_TYPE"]
+
+			part.SizeInBytes, err = requiredUdevPropUint(udevProps, "ID_PART_ENTRY_SIZE")
+			if err != nil {
+				return fmt.Errorf("cannot get required udev property for device %s (a partition of %s): %v", partDev, d.Dev(), err)
```

Comment:
minor: store this error message somewhere, or use a nested function to emit it.

Reply:
Hmm maybe group those into something like:
```
fillRequiredPropsFromUdev := func() error {
    // fill from ID_PART_ENTRY_SIZE, ID_PART_ENTRY_NUMBER, ID_PART_ENTRY_OFFSET, 
    // ID_PART_ENTRY_UUID, MAJOR, MINOR, 
}
if err := filleRequiredPropsFromUdev() ; err != nil {
    return fmt.Errorf("cannot get required udev properties for device ...", partDev, d.Dev())
}
```

Reply:
I think I implemented the requested suggestion here